osu.Game/Online/API/APIRequest.cs

```diff
@@ -88,7 +88,12 @@ public void Perform(IAPIProvider api)
             if (checkAndScheduleFailure())
                 return;
 
-            API.Schedule(delegate { Success?.Invoke(); });
+            API.Schedule(delegate
+            {
+                if (cancelled) return;
```

Comment:
`checkAndScheduleFailure` ? Should still fire the `Failed` event imo.

Reply:
Is called on line 114, should be enough.